File: csharp/Tennis/TennisGame6.cs
namespace Tennis;

public class TennisGame6 : ITennisGame
{
    private int player1Score;
    private int player2Score;
    private string player1Name;
    private string player2Name;

    public TennisGame6(string player1Name, string player2Name)
    {
        this.player1Name = player1Name;
        this.player2Name = player2Name;
    }

    public void WonPoint(string playerName)
    {
        if (playerName == player1Name)
            player1Score++;
        else
            player2Score++;
    }

    public string GetScore()
    {
        string result;

        if (player1Score == player2Score)
        {
            // tie score
            string tieScore;
            switch (player1Score)
            {
                case 0:
                    tieScore = "Love-All";
                    break;
                case 1:
                    tieScore = "Fifteen-All";
                    break;
                case 2:
                    tieScore = "Thirty-All";
                    break;
                default:
                    tieScore = "Deuce";
                    break;
            }

            result = tieScore;
        }
        else if (player1Score >= 4 || player2Score >= 4)
        {
            // end-game score
            string endGameScore;

            switch (player1Score - player2Score)
            {
                case 1:
                    endGameScore = $"Advantage {player1Name}";
                    break;
                case -1:
                    endGameScore = $"Advantage {player2Name}";
                    break;
                case >= 2:
                    endGameScore = $"Win for {player1Name}";
                    break;
                default:
                    endGameScore = $"Win for {player2Name}";
                    break;
            }

            result = endGameScore;
        }
        else
        {
            // regular score
            string regularScore;

            var score1 = player1Score switch
            {
                0 => "Love",
                1 => "Fifteen",
                2 => "Thirty",
                _ => "Forty"
            };

            var score2 = player2Score switch
            {
                0 => "Love",
                1 => "Fifteen",
                2 => "Thirty",
                _ => "Forty"
            };

            regularScore = $"{score1}-{score2}";

            result = regularScore;
        }

        return result;
    }
}

File: php/src/TennisGame6.php
<?php

namespace TennisGame;

class TennisGame6 implements TennisGame {
    private $player1Name;
    private $player2Name;
    private $player1Score = 0;
    private $player2Score = 0;

    public function __construct($player1Name, $player2Name)
    {
        $this->player1Name = $player1Name;
        $this->player2Name = $player2Name;
    }

    public function wonPoint($playerName): void
    {
        if ($playerName === "player1") {
            $this->player1Score++;
        } else {
            $this->player2Score++;
        }
    }

    public function getScore(): string
    {
        $result = "";

        if ($this->player1Score == $this->player2Score) {
            // tie score
            switch ($this->player1Score) {
                case 0:
                    $result = "Love-All";
                    break;
                case 1:
                    $result = "Fifteen-All";
                    break;
                case 2:
                    $result = "Thirty-All";
                    break;
                default:
                    $result = "Deuce";
                    break;
            }
        } else if ($this->player1Score >= 4 || $this->player2Score >= 4) {
            // end-game score
            if ($this->player1Score - $this->player2Score == 1) {
                $result = "Advantage " . $this->player1Name;
            } else if ($this->player1Score - $this->player2Score == -1) {
                $result = "Advantage " . $this->player2Name;
            } else if ($this->player1Score - $this->player2Score >= 2) {
                $result = "Win for " . $this->player1Name;
            } else {
                $result = "Win for " . $this->player2Name;
            }
        } else {
            // regular score
            switch ($this->player1Score) {
                case 0:
                    $score1 = "Love";
                    break;
                case 1:
                    $score1 = "Fifteen";
                    break;
                case 2:
                    $score1 = "Thirty";
                    break;
                default:
                    $score1 = "Forty";
                    break;
            }
            switch ($this->player2Score) {
                case 0:
                    $score2 = "Love";
                    break;
                case 1:
                    $score2 = "Fifteen";
                    break;
                case 2:
                    $score2 = "Thirty";
                    break;
                default:
                    $score2 = "Forty";
                    break;
            }
            $result = $score1 . "-" . $score2;
        }

        return $result;
    }
}
